apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/Schema.kt

```diff
@@ -92,6 +91,15 @@ class Schema(
     }
   }
 
+  /**
+   * Returns whether the `typePolicy` directive is present on at least one object in the schema
+   */
+  fun hasTypeWithTypePolicy(): Boolean {
+    return typeDefinitions.values.filterIsInstance<GQLObjectTypeDefinition>().any { objectType ->
```

Comment:
Note for our future selves: we'll want to support `@typePolicy` on interfaces and unions (see https://github.com/apollographql/apollo-kotlin/issues/3356). This will need to be updated when this happens